src/components/footer/config/factory.ts

```diff
@@ -0,0 +1,127 @@
+import { Language } from '@smg-automotive/i18n-pkg';
+
+import { Link, LinkConfig, LinkTargets } from './link';
+import { Environment } from '../../../types/environment';
+import { Brand } from '../../../types/brand';
+
+import { FooterConfigInterface } from '.';
+
+type LocalizedLinks = Record<Language, string>;
+
+export interface LinkInstance {
+  translationKey: string;
+  link?: LocalizedLinks;
+  target?: LinkTargets;
+  isVisible: boolean;
+  onClick?: () => void;
+}
+
+interface LinkSectionConfig {
+  title: LinkConfig[];
+  items: LinkConfig[];
+}
+
+export interface FooterConfigSection {
+  title: LinkInstance[];
+  items: LinkInstance[];
+}
+
+export interface FooterConfigInstance {
+  sections: FooterConfigSection[];
+  apps: {
+    android: LinkInstance[];
+    apple: LinkInstance[];
+  };
+  socialMedia: {
+    facebook: LinkInstance[];
+    instagram: LinkInstance[];
+    twitter: LinkInstance[];
+    youtube: LinkInstance[];
+  };
+  companies: LinkInstance[];
+}
+export class FooterConfig {
+  brand: Brand;
+  config: FooterConfigInterface;
+  environment: Environment;
+  useAbsoluteUrls: boolean;
+  mappedConfig?: FooterConfigInstance;
+  linkProtocol: string;
+  domains: Record<Brand, Record<Environment, string>>;
+
+  constructor({
+    config,
+    brand,
+    environment = 'production',
+    useAbsoluteUrls = false,
+  }: {
+    config: FooterConfigInterface;
+    brand: Brand;
+    environment?: Environment;
+    useAbsoluteUrls?: boolean;
+  }) {
+    this.brand = brand;
+    this.config = config;
+    this.environment = environment;
+    this.useAbsoluteUrls = useAbsoluteUrls;
+    this.domains = {
+      as24: {
+        production: 'www.autoscout24.ch',
+        preprod: 'int.autoscout24.ch',
+      },
+      ms24: {
+        production: 'www.motoscout24.ch',
+        preprod: 'int.motoscout24.ch',
+      },
+    };
+    this.linkProtocol = 'https';
+  }
+
+  getMappedConfig(): FooterConfigInstance {
+    if (this.mappedConfig) return this.mappedConfig;
```

Comment:
I am not sure I get this line right. Does that mean that, if a `mappedConfig` is specified, we can have a (more or less) custom Footer?

Reply:
Nice catch, I think I ended up solving it differntly

Reply:
So not quite. It's not related to or intended at this stage to pass any configuration for the items to the footer. The footer object is mapping the structure updating the format with visibility settings and absolute and relative URLs. The result of that operation ist at the end stored to the current instance of the object `this.mappedConfig = mappedConfig;` what this code does is ensure the mapping operations are only performed a single time per instance. So if we wanna get the configuration in another place we don't need to recalculate it but can just pass back the result.

```ts
 const footerConfigInstance = new FooterConfig({
      config: footerConfig,
      brand,
      environment,
      useAbsoluteUrls,
    });
return footerConfigInstance.getMappedConfig();
```

The MDN docs on classes might help you get a better understanding https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Objects/Classes_in_JavaScript

Reply:
Ah ok! I get it now, thank you